dask_kubernetes/core.py

```diff
@@ -379,6 +379,8 @@ def scale_up(self, n, pods=None, **kwargs):
         --------
         >>> cluster.scale_up(20)  # ask for twenty workers
         """
+        if 'kubernetes-maximum-workers' in config:
+            n = min(n, config['kubernetes-maximum-workers'])
```

Comment:
Not sure what logging is like in dask-kubernetes, but logging that the number of workers is being limited by the config seems like it could save someone some time in the future.